pkg/httphelper/httphelper.go

```diff
@@ -6,14 +6,30 @@ import (
 	"net/http"
 )
 
-func Error(w http.ResponseWriter, err error) {
+type JSONError struct {
+	Code    int         `json:"code"`
+	Message string      `json:"message"`
+	Detail  interface{} `json:"detail,omitempty"`
```

Comment:
Okay, yeah, given that this will be used on both client and server end of things, it's best to keep the complexity at the narrow end and use `json.RawMessage`.